### ISSUE TITLE
Video fails to play beyond 2 minutes

### ISSUE THREAD

**Issue by Stefanperlarsson** [CastagnaIT/plugin.video.netflix#1678]
### Netflix add-on version

1.23.2+matrix1

### Operative systems used

Android

### Kodi version used

Kodi 20 (Nexus)

### Description of the bug

Since last week there seem to be an issue playing videos beyond 2 minutes.
The video stops playing, audio seem to continue however.

Starting the video half-way through also has the same issue.
I tried multiple video streams with this as well, but no change.

Potentially related to https://github.com/xbmc/inputstream.adaptive/issues/1461
but I am unable to find a .zip to install the newer version of inputstream with.

If one is provided i can help test it.

Thank you.

### Steps to reproduce the behavior

1. Navigate to any video in Netflix
2. Play the using any video stream
3. Wait ~2 minutes
4. See the video stop and fail to proceed

### Debug log - mandatory

I seem to be unable to get the kodi.log file since I am using a chromecast. Please advise on how to get it and I will try my best.

**Comment by tehjul** [comment]
Same bug here on a Nvidia Shield TV 2017. After 2 minutes, the video freezes and the audio still playing.
I think it's more Netflix related as this bug appeared in the same time for me as my Netflix app is only playing at 720p event if I have 800Mb of bandwidth. 3 or 4 days ago I didn't have any issue on both.

**Comment by CastagnaIT** [comment]
this is duplicate of already opened issue #1675

**Comment by CastagnaIT** [comment]
i copy here my previous investigations:

the 2 minuts playback its normal because the first 2 minuts are unencrypted,
and so after that time the DRM cannot do the decryption and playback its ofc broken.

i tried investigate but i have zero ideas of whats changed on netflix servers...
website data looks the same as before, but device drm result in a weird error relative to initialize the drm session
i have not found any info about this error on the web
```
02-04 10:54:50.228  3364  3406 E oemcrypto_cenc: OEMCrypto_LoadDRMPrivateKey: error in te_launch_operation: 0x1e
02-04 10:54:50.229  5400  5834 W System.err: android.media.NotProvisionedException: Failed to open session
```

**Comment by nroets1** [comment]
First the important. Thank you very much for all working on this amazing addon. Your efforts are appreciated.

I can confirm that on Android MiBox S 4K and another Android TV box (HAKU), only 540p playback is possible. This is with kodi 21 omega beta 2 and latest inputstream update.

On my Linux pc with Ubuntu 22.04. Kodi 20.3, 1080p playback is working without an issue.

**Comment by josefede** [comment]
Amazon Firestick here. Also only first 2 minutes playback at 1080p and 4K

**Comment by wajo666** [comment]
Philips 58PUS7304/12 Android 9
Kodi 21 beta2
Inputstream 21.4.2

same issue

**Comment by PaulHeyworth** [comment]
I can confirm I have the same issue too. On FireCube GEN2, GEN3, Firestick 4K and Nvidia Shield 2017.  I've tried Kodi 20 Nexus, but also Omega Beta 2 and all exhibit the same behaviour of anything above 540p (based on the Netflix add-on settings) freezing after 2 minutes.  Only the video stream is impacted and the audio continues to play unhindered.

Hopefully there is a fix, because the add-on has been excellent in my opinion and a much better experience than the native app.

Thanks for your help.

Paul.

**Comment by tehjul** [comment]
Hello guys, someone tried the official android app ? On my shield the image quality is poor (looks like 720p) even if I have about 800Mb/s bandwidth when I test my connections for in the app. Maybe it's Netflix related

**Comment by nroets1** [comment]
@tehjul, I have tried on both my Android tv boxes, MiBox S 4K and HAKU, with the native netflix app. On both the playback is 4K without any issue. I have used the Google Home remote on my phone with hackers keyboard installed. During playback, press F8 to see the playback resolution. I have a suspicion, and I hope I'm worng, that Netflix now only allow their native android app to decode beyond 540p on Android......

**Comment by RealJustMe** [comment]
@tehjul I noticed the same. The LG app has a clear image, but the app on my Fire TV looks like 720p.

**Comment by tehjul** [comment]
@nroets1 You're right I think. I plugged a keyboard and pressed F8 and I saw that I'm stuck at 540p (only in the native app, in Kodi it's ok but there's still this 2 minutes bug). I tried on my other devices (Computer, tablet, ..) and I have 2160p/1080p without any issue. Where did you find that information about android devices ?

**Comment by nroets1** [comment]
@tehjul , I did not find it anywhere. Just assuming / guessing. As I said, I hope I'm wrong.

**Comment by tsioukas** [comment]
@nroets1, Correct me if I didn't understand it. You say that using Netflix Google TV APP you get 540p? Did you know if they ban your native ESN number? Have you ever used this Netflix/Device ESN number in KODI Netflix Addon in the past?

**Comment by nroets1** [comment]
@tsioukas , no, I get 4k playback on the Android tv Netflix app. With kodi addo  I only get 540p to work without freezing after 2 minutes.

**Comment by grapenose** [comment]
I had no luck with Sony TV other than playing at 540.
I assumed that I would have the same problem on my Fire Cube Gen3 .This runs android 11 , I think.
I tried  casting from my PC laptop which worked, but is a bit of a pain.
As a last resort I installed on the Fire Cube and it is working, with what appears to be HD.
Go figure!

**Comment by grapenose** [comment]
Now working on my Sony TV.

**Comment by Ferosmyth** [comment]
Same issue Sony android Tv, logged out, uninstalled the add on, installed again, now it doesn´t allow me to log in even with the authentication key. Help please!

**Comment by BBican63** [comment]
> Now working on my Sony TV.

**grapenose** can you explain to me how it works for you?
My Sony BRAVIA TV still works for 2 minutes only.

**Comment by PaulHeyworth** [comment]
I'm a little puzzled by the lack of noise on this issue, because it seems now that the add-on is completely broken on Android devices, unless you force playback at 540p.  The video simply stops after 2 minutes, making it no longer usable.  

Am I off the mark with this statement and is it only affecting some users?  Are there still people able to use the Kodi with the addon on Android devices?  If so, I'll resume looking at local issues.

**Comment by lukaszkuch** [comment]
I want to confirm that it's not working for me either on Android TV. I've changed configuration to force 540p and some files starts to work, but others doesn't start at all.

**Comment by grapenose** [comment]
> > Now working on my Sony TV.
> 
> **grapenose** can you explain to me how it works for you? My Sony BRAVIA TV still works for 2 minutes only.
Have had no power since Monday due to a massive storm that ripped through Victoria, Australia. 
So first chance to check and Netflix is not working on Sony or Fire cube other than at 540. 
Can't play One Day at all( get error message) but can play other series such as Fool Me Once at 540

**Comment by kokobokoroko** [comment]
Mi box working only 540 p
Android 12
Plan netflix -uhd

**Comment by pmsobrado** [comment]
Fire Stick 4K Max here, same problem, with 4K and HDR, no playback, only sound and only some times, and the "nf_is_ads" or something like that "MSPlayback" error (I don't remember the error names).

**Comment by manolischania** [comment]
i have a question for you who have problem .  netflix account is from your country or some other country ?

**Comment by tsioukas** [comment]
Our country.

- The problem, at start. affect both linux and android devices. With recent update of input stream adaptive it looks like that linux problem solved. 
- The nature of problem looks like is on decoding of DRM content (above 540p). First 2 minutes doesn’t have any DRM. After that audio played normally but without video (freezes or get black)

From the above I believe that it’s an input stream problem and not Netflix addon problem (if it’s using the correct decryption key, as it do it before)

The problem affects both for 21.x beta (2 or 3) and 20.3 versions of Kodi. I am waiting to see if something fixed on 20.4 but at this time isn’t ready for Android devices.

I don’t know if @CastagnaIT find the time to look it again.  Stefano do we have any update about this?

Does anyone reported this issue on input stream adaptive git?

**Comment by CastagnaIT** [comment]
people are making a huge mess of comments on all the threads saying a lot of nonsense things and also ignoring my few comments, I'm a little tired of repeating things all the time, here i summarize:

- first 2 minutes of video are unencrypted and so correctly played, this is known normal behaviour, since you cant decrypt you cant play more than 2 minutes
- different platform, different handling on client and also server side (website server), this is the worst complicated control system netflix uses to limit everything most of time ESN linked. So android =/= linux =/= windows. Generalizing everything by comparing playback on different platforms is definitely wrong and useless
- Nothing has been changed on ISAdaptive despite recent updates (that i published myself), and so NO fixes has been done on ISAdaptive for linux in specific, also because ISA decrypter share the same source code of Windows, if there are differents behaviors blame netflix website
- there seems to be a problem on the stream encryption keys that can cause CryptoException on HD/FULLHD streams, but this error seem not always present on logcat, this confuse me
- there seems to be an error opening CDM session but it is possible that it is a consequential cause of the previous
- comparing old manifest response (of many months ago) with new one, i havent found notable differences
- if it is a decrypter key issue it is possible that the required license does not provide them, its a possibility but then if you play a video on webbrowser on Linux should result at same 540p resolution max, if not so, there may be differences on MSL manifest/license request that i don't know about or other things related

Since nothing has changed on kodi, ISAdaptive and on the addon, something unclear is changed on netflix website server
it is possible that new changes are needed on ISA, but seems more like something related to website quirks but i could be wrong since there are still DRM things that i dont know exaclty how works given poor docs and his complexity

On my part i currently have no other information and no other solutions
And i don't even know who it would help on this

**Comment by tsioukas** [comment]
@CastagnaIT, sorry that I mess your threads. I will never comment again even to report my devices behavior. I thought that Linux problem solved by ISA latest update as users reported (and that’s why issue closed). I hope to find the solution.

**Comment by pmsobrado** [comment]
> i have a question for you who have problem . netflix account is from your country or some other country ?

Mine is from other country, but the NF app works fine on the same device and I don't think that's the issue here.

**Comment by manolischania** [comment]
> > i have a question for you who have problem . netflix account is from your country or some other country ?
> 
> Mine is from other country, but the NF app works fine on the same device and I don't think that's the issue here.

I didn't say that this is the problem, I just asked a question because we have strange behavior in different accounts, I am from Greece and with a Turkish account it stops every two minutes from a Greek account and a Greek ip does not stop

**Comment by pmsobrado** [comment]
Are you saying that a greek account with a greek IP plays well on Kodi and Android beyond 540p? HD and UHD?

**Comment by manolischania** [comment]
> Are you saying that a greek account with a greek IP plays well on Kodi and Android beyond 540p? HD and UHD?

on certified machines like nokia and 8010 with greek ip and greek account plays 4k dolby vision also with nigerian account and greek ip ... not with tourkish account an greek ip ...

**Comment by pmsobrado** [comment]
Interesting...

**Comment by BBican63** [comment]
I noticed a very strange thing on the original netflix app. I have a Sony BRAVIA XR-65A75K TV, Croatian account and Croatian IP. 
When I press the info button on the remote (the button below the number 7 button), I get the technical details of the currently playing video and it says "Video: 0.55 Mbps / 540 / Codec DV5" (image attached). The movie I'm watching is declared as 4K. The picture quality is excellent, at least HD. If I choose a movie that is declared as HD the situation is very similar "Video: 0.xx Mbps / 540 / Codec ...". 
Can someone check my statements?
![Video](https://github.com/CastagnaIT/plugin.video.netflix/assets/160004709/f02833bc-2e73-495a-ba91-285b1ed3ea0a)

**Comment by BBican63** [comment]
Sorry, I forgot to mention. Before this issue it always said "Video: 27.xx Mbps / 2160 / Codec ..." or something similar.

**Comment by krzychuw88** [comment]
![PXL_20240216_105307353](https://github.com/CastagnaIT/plugin.video.netflix/assets/43517101/4a7752dd-b05e-44be-a9a4-4df327aa1c00)
I have the same issue. Even factory reset doesn't help. On different netflix account also low quality playback.

**Comment by wajo666** [comment]
I think, netflix changed drm security. It is for  security with sharing. He started blocked netflix app and you need to be in one home. Only playing via web browser is not secure under home check. Could someone check which version of video is played via web browser?

**Comment by BBican63** [comment]
web browser resolution for me is full HD 1920x1080, but that's because my laptop supports a maximum of 1920x1080. I assume the resolution would be 4K if my laptop supported it. It would be good if someone else could try by pressing Alt+Crtl+Shift+D on the keyboard during playback.
![Video_web](https://github.com/CastagnaIT/plugin.video.netflix/assets/160004709/f507dd8f-8993-478f-9676-b7309bd07715)

**Comment by BBican63** [comment]
InputStream Adaptive 20.3.18 February 21, 2024 appeared, but my problem remained the same. Limit of 2 minutes and maximum resolution of 540p.

**Comment by grapenose** [comment]
Netflix seems to be working again for me.
Did change over to USA from Australia with a DNS, so maybe it won't last.

**Comment by josefede** [comment]
> Netflix seems to be working again for me. Did change over to USA from Australia with a DNS, so maybe it won't last.

That's interesting. Some people seem to be able to get it to work with certain account and IP locations. I'm assuming it does not work in your case if you don't use a VPN?

**Comment by josefede** [comment]
> Netflix seems to be working again for me. Did change over to USA from Australia with a DNS, so maybe it won't last.

That's interesting. Looks like it works for some people depending on their location. Does it still work if you disable the DNS?

**Comment by grapenose** [comment]
Now I really am confused.
Does not work on my Amazon Fire Cube other than at 540.
Works on my Sony Bravia TV with or without the DNS.

**Comment by BBican63** [comment]
My Sony BRAVIA XR-65A75K (location Croatia) is still not working. It only works at 540p.

**Comment by grapenose** [comment]
Bravia H series 2020, Android 10.
Still working

**Comment by etgecata** [comment]
Sorry for the bad english. My guess is that Netflix limits Android devices to 540p. I have an Nvidia Shield 2019. I have never used its esc number in the Kodi addon. In the official Netflix app, my resolution is 540p. I have 4k in the Netflix app on the Samsung TV.
There are two threads on the Nvidia Shield forum for other users that confirm this:

https://www.nvidia.com/en-us/geforce/forums/shield-tv/9/537184/netflix-unable-to-output-anything-higher-than-540p/
https://www.nvidia.com/en-us/geforce/forums/shield-tv/9/537504/netflix-bitrate-low-on-hd-streams-on-shield-tv-pro/

**Comment by pmsobrado** [comment]
But the official Netflix app seems to work fine for me, it's only the plugin the one that doesn't go beyond 540p :(

**Comment by PaulHeyworth** [comment]
I have numerous Android devices, including FireCube Gen2, Gen3, 4K Firestick, and a Shield 2017.  The native Netflix app plays 1080p and 4K fine.  Additionally, before late January the Netflix Kodi app would output 4K on all the Amazon devices, but 1080p only on the Shield.

Since late January, I have to set 540p in the Netflix Kodi add-on, otherwise the video output freezes after 2 minutes, while audio continues fine.

I have no such issues with the native Neflix app. It's a shame, because I thought the Kodi add-on was excellent, but it's not not usable on a relatively big screen, with such a low resolution.

**Comment by pmsobrado** [comment]
OK, the official app only outputs 540p max on my Fire TV as well :(

**Comment by pmsobrado** [comment]
I tried that with my BT keyboard and worked on the Fire TV and Android TV app, but not on my phone. Tried almost all hotkeys, nothing. That app is not compatible with my phone, it seems.

**Comment by cichytom** [comment]
It works with all resolutions on the Sony TV from 2023 (Android TV 10, security patch level of 01.09.2023).
On Google Chromecast 4.0 4k (Android TV 12, security patch level dated 01.01.2024) at resolutions above 960x540 (e.g. 1280x720) it stops after 2 minutes. At resolutions of 768x432, 960x540 it works without problems.

**Comment by josefede** [comment]
> It works with all resolutions on the Sony TV from 2023 (Android TV 10, security patch level of 01.09.2023). On Google Chromecast 4.0 4k (Android TV 12, security patch level dated 01.01.2024) at resolutions above 960x540 (e.g. 1280x720) it stops after 2 minutes. At resolutions of 768x432, 960x540 it works without problems.
> 
> In Kodi 20.3 on the Sony TV from 2023 (Android TV 10) I can set a maximum resolution of 1920x1080p. I do not have a choice of 4K resolution.

It seems Android TV version may be significant here then. Maybe more people can confirm this. I can't though

**Comment by etgecata** [comment]
![Nvidia Shield Pro 2019](https://github.com/CastagnaIT/plugin.video.netflix/assets/17301428/2d85a0a0-bdf8-4603-aabd-f63d05ff9bb7)
![PS 5](https://github.com/CastagnaIT/plugin.video.netflix/assets/17301428/03c358f3-e70c-499d-afc5-d99e527a4edb)
![Samsung TV](https://github.com/CastagnaIT/plugin.video.netflix/assets/17301428/23571b07-504e-487c-87b6-6fdd89b30e00)
![SONY TV](https://github.com/CastagnaIT/plugin.video.netflix/assets/17301428/f3384d98-d930-4936-93f0-1a36d77716e1)

This is beyond scandalous to me. I'm sure Netflix is intentionally hurting Android subscribers with 540p. Here are pictures of all the devices I have at home from the official Netflix app.
You pay a subscription for 4k - you get 540p if you watch on Android TV.
Samsung TV and PS5 - 2160 p
Nvidia Shield 2019 and Sony TV-540 p

**Comment by PaulHeyworth** [comment]
@etgecata This is not the experience I'm getting with multiple Android devices.  Due to the low resolution the Netflix Kodi add-on is now restricted to (540p), I've had to switch back to the native Netflix app, on multiple Android device types, and I'm getting 4K just fine.

The devices I have are:-

- FireCube Gen3
- FireCube Gen2
- Firestick 4K
- Firestick (limited to 1080p of course)
- Shield 2017

**Comment by etgecata** [comment]
For me, it is as I wrote it and attached the photos. It may also depend on the country, I don't know. I'm from Bulgaria.

**Comment by Paco8** [comment]
Did you use the Netflix add-on on those devices restricted to 540p?

My Shield and my Chromecast with Google TV were banned and restricted to 540p because I used the ESN in the Netflix add-on. After 6 months of not using either the Netflix app or the Netflix add-on, my Chromecast has been unbanned, and I can get 1080p again in the Netflix app. However, the Shield is still banned; the difference is that I continued to use the Netflix add-on on this device.

**Comment by BBican63** [comment]
As I said before, OFFICIAL NETFLIX app on Sony BRAVIA XR-65A75K (location Croatia) still 540p. You can see the screenshot in my previous message. My neighbor uses "TCL Android TV-QLED-4K-UHD" and his OFFICIAL NETFLIX app works at 2160p 4K. So I think the location is not the problem, but the version of Android or the hardware of the device. Why the Sony Bravia doesn't do 4K, I don't know.

**Comment by etgecata** [comment]
I've written it before.
I have not used an ESN in the Netflix add-on.
I've had the Nvidia Shield 2019 for a few months now and haven't used ESN in the add-on.

**Comment by etgecata** [comment]
I am absolutely certain of one thing. Netflix is brutally harming Android TV subscribers. Maybe not all of them, I don't know. But from what I saw on the Internet, there are many. This is scandalous to me.

**Comment by kokobokoroko** [comment]
as far as i understand everything is in the hands of netflix


На пт, 1.03.2024 г., 12:45 ч. etgecata ***@***.***> написа:

**Comment by paulison** [comment]
Hi, it seems that the Netflix Kodi Add-on works fine for a while, after either first installing it or if it is used for the first time in a little while, but only for a short time.

I have several devices running the Netflix Add-on on Kodi, various Fire TV devices, Nvidia Shield, Mecool KM2, and have found that once used it will stop working properly the following day, and you have to limit the Video quality to 540p for it to play anything beoynd 2 minutes.  I can only think that somehow Netflix can track that the service is being used via Kodi and is limiting it to 540p once they have found it, maybe using the ESN but I doubt it as I've changed this, reset it in teh Add-on, but it seems to make no difference.

**Comment by Bloushiiii** [comment]
Nvidia shield Tv Pro 2017 and same problem.. Native app + Kodi both stuck at 540p. After logout on Kodi Netflix add-ons cant login again.

PS5 plays perfect 4k Netflix so it's sad to see Kodi stopped working coz app experience is better on Kodi .. Hope we get a fix soon..

**Comment by BBican63** [comment]
Bloushiiii - To re-login to netflix Kodi addon:
Edit file /.kodi/addons/plugin.video.netflix/resources/lib/services/nfsession/session/access.py
Changed line 127 in the access.py file from this:
email = email_match.group(1).strip() if email_match else None
to this:
email = 'your netflix account e-mail address'
It's works for me.

**Comment by Bloushiiii** [comment]
> Bloushiiii - To re-login to netflix Kodi addon: Edit file /.kodi/addons/plugin.video.netflix/resources/lib/services/nfsession/session/access.py Changed line 127 in the access.py file from this: email = email_match.group(1).strip() if email_match else None to this: email = 'your netflix account e-mail address' It's works for me.

Yes bro but what's the use if it's not 4K .... I will wait until we get all the options back like before...

**Comment by adambomber** [comment]
Hi, I believe I've been suffering the same 540p issue since trying the Kodi addon on my Shield TV Pro. I've been going nuts trying different streaming devices to see if there was just something wrong with my Shield, and so far, most of them have the same issue regardless of OS. I'm kind of at a loss on what to do, since this sounds like something Netflix is doing on an ESN or account basis.

**Comment by PaulHeyworth** [comment]
I noticed an interesting change with this earlier today.  In order to get past the dreaded 2 mins playback, I've previously had to force the playback to 540p in the Netflix add-on expert settings.   However, I now have it set to 4K, forcing it to ask for the preferred video quality at playback.  Regardless of having the maximum set to 4K, it only now offers the option of playing 540p and nothing above this.

This is another change in behaviour, seemingly without anything changing in Kodi or the Netflix add-on.  My complete guess it that this is deliberate by Netflix and they are trying to lock down use of their service.  Interestingly, I see they have had a record surge in subscribers since locking down password sharing.

**Comment by Ferosmyth** [comment]
> Bloushiiii - To re-login to netflix Kodi addon: Edit file /.kodi/addons/plugin.video.netflix/resources/lib/services/nfsession/session/access.py Changed line 127 in the access.py file from this: email = email_match.group(1).strip() if email_match else None to this: email = 'your netflix account e-mail address' It's works for me.

This helped me a lot, thanks!
Still, only 540p reproduction allowed, it is Netflix messing with Kodi Addon

**Comment by Bloushiiii** [comment]
I know this will work but until it's 4k I won't b using Kodi Netflix add-ons. 540p looks so bad ...

**Comment by BBican63** [comment]
KODI Netflix addon on Android devices is no longer talked about. Does that mean it's over ?
Another thing.
I know a lot of people have this problem. On my Sony Bravia  Google TV, the Netflix official app only plays movies in 540p. It is interesting that the trailers are played in 2160 (4K). It all started a month ago when the KODI NETFLIX addon stopped working. Is that a punishment ? Have we been put on some kind blacklist ? Does the blacklist get deleted over time ?

**Comment by grapenose** [comment]
Kodi Netflix still works on my Sony Bravia at more than 540.
As I reported before I have a DNS and have set it to USA. I'm in Australia and used to share a sub, this stopped working on the android TV version but worked on Kodi for a while.
It seems if I switch between between USA and Australia with my DNS the Kodi app continues to work.
It does not work with my Amazon Cube 3rd gen.
Has anyone had a similar experience?

**Comment by nroets1** [comment]
Today I've noticed an update in kodi on my MiBox 4K for inputstream adaptive. I thought let's try netflix addon again. It now plays back 1080p (my tv is only 1080p) flawlessly without the 2 minute issue. I'll test later on my 4k tv with Hako android tv box and report back.

**Comment by paulison** [comment]
> Today I've noticed an update in kodi on my MiBox 4K for inputstream adaptive. I thought let's try netflix addon again. It now plays back 1080p (my tv is only 1080p) flawlessly without the 2 minute issue. I'll test later on my 4k tv with Hako android tv box and report back.

I can also confirm that the Netflix Kodi addon is working perfectly again, tested on several Fire TV devices, Android & Android TV boxes.  The Inputstream Adaptive update seems to have resolved the problem

**Comment by krzychuw88** [comment]
Working only 1080p

**Comment by adambomber** [comment]
I'm glad to hear the issue seems to be resolved.

I think I realized my Netflix issue wasn't related to Kodi after all, but
to one 1080p show having its bitrate locked by Netflix to 0.85 mbps on my
Shield for some reason (my LG TV's native app plays it fine). I don't know
who to complain to hah

**Comment by DAndersen81** [comment]
On my shield unfortunately I still have the 2 minute bug. The inputstream adaptive has updated, I've tried restarting Kodi, rebooting, but no luck.
Are there any specific settings maybe I should check?

**Comment by josefede** [comment]
> On my shield unfortunately I still have the 2 minute bug. The inputstream adaptive has updated, I've tried restarting Kodi, rebooting, but no luck. Are there any specific settings maybe I should check?

Do you use Kodi Nexus or Omega? AFAIC the InputStream Adaptive patch has only been released for Omega.

Also doesn't work for me because I'm using Nexus

**Comment by DAndersen81** [comment]
> > On my shield unfortunately I still have the 2 minute bug. The inputstream adaptive has updated, I've tried restarting Kodi, rebooting, but no luck. Are there any specific settings maybe I should check?
> 
> Do you use Kodi Nexus or Omega? AFAIC the InputStream Adaptive patch has only been released for Omega.
> 
> Also doesn't work for me because I'm using Nexus

I'm using omega... The inputstream adaptive plugin updated already.

**Comment by CastagnaIT** [comment]
there are no changes on ISA about the decrypter so i don't understand how it should have "solved" the problem,
this is more a coincidence, it depends on the n€flix website server
maybe they are finally trying to solve the problem?! IDK...

**Comment by CastagnaIT** [comment]
i have tried closing my account for a month, to see if my account web page change with the new interface
for the "auth" key login problem, yesterday i tried reopen with new subscription but same accound webpage and the "auth" key login still works as expected, then i have no idea what happens on accounts that have this "new" interface

if someone want that i try to add a kind of fix instead to hack the addon,
you have to send to me the html page, this can be easily obtained by coping the output of
`page_response` var from `login_auth_data` method in access.py

**Comment by nroets1** [comment]
> there are no changes on ISA about the decrypter so i don't understand how it should have "solved" the problem, this is more a coincidence, it depends on the n€flix website server maybe they are finally trying to solve the problem?! IDK...

You are right. I've tried on my other Android tv box (Hako) and problem still exists. On my MiBox 4k it's working. Both use omega rc2 with latest ISA addon. It seems like an intermiitent problem from the Netflix side.

**Comment by etgecata** [comment]
I was thinking of canceling my Netflix subscription. I had put my subscription on pause for a month. But after reading here that there is a change today I reactivated the subscription. Unfortunately, there is no change in the supplement for me. I can't play 4K content at all. 1080 plays for 2 minutes. I tried on Nvidia Shield pro 2019, Nvidia Shield 2015, Sony TV. I have Kodi 21 Omega installed on all devices. If it matters, I'm from Bulgaria.

**Comment by Bajjmm** [comment]
And when will we see the InputStream Adaptive update for Kodi Nexus (android Sony smart TV)?

**Comment by pmsobrado** [comment]
> And when will we see the InputStream Adaptive update for Kodi Nexus (android Sony smart TV)?

I would also like to know this :S

**Comment by grapenose** [comment]
Just updated Kodi to Omega on my Amazon Cube and some videos now play in HD. However I'm still getting a message that some are not available ie3 Body Problem, The Gentlemen. 
These play on my Sony Bravia in HD

**Comment by MauriceW67** [comment]
> but if that page dont have the email anymore, then you will have to point to me a better website endpoint/page with his html

I just checked and my e-mail is now visible on the main "YourAccount" page again. A few months ago it was only visible on the sub-page called "Security" I believe. So I guess Netflix has been playing with the layout again.

I didn't try logging on yet with the orignal access.py. Will let you know later.

**Comment by NoneCreator** [comment]
hi all,

so it perfectly works for me, inclduing true 4k and DV.
I upgraded Kodi to 21 Omega a couple of minutes ago.
According to previous posts here that shouldn't fix it.
However, a couple of days back I also installed the "InputStream Helper (Slyguy Wrapper)" from the "SlyGuy InputStream Helper Repository". Maybe that did the trick.
Anyway, works perfectly for me.

I'm running Kodi Omega on an NVIDIA Shield (2019) with said SlyGuy addon.
Hope this helps you good folken as well.

**Comment by t4e-dev** [comment]
On FireTV Stick and FireTV the issue is, at least for me, not fixed.
InputStream Adaptive works only for 640p max.

**Comment by NoneCreator** [comment]
> On FireTV Stick and FireTV the issue is, at least for me, not fixed. InputStream Adaptive works only for 640p max.

did you try the SlyGuy "InputStream Helper (Slyguy Wrapper)".

**Comment by etgecata** [comment]
I installed "InputStream Helper (Slyguy Wrapper)" from the "SlyGuy InputStream Helper Repository".
Nvidia Shield Pro 2019- Kodi Omega
There is no improvement.

**Comment by NoneCreator** [comment]
so yeah, as of today, it's not working anymore

**Comment by josefede** [comment]
> so yeah, as of today, it's not working anymore

Definitely something on their side. Was good while it lasted

**Comment by MauriceW67** [comment]
> so yeah, as of today, it's not working anymore

Can confirm, doesn't work anymore So strange. Don't understand what Netflix is doing ☹️

**Comment by giddybitty** [comment]
@CastagnaIT , do you think there is a chance you can add an option to the addon, to use the Netflix app to watch content?
That way we‘d still be able to export stuff to the Kodi library.
However, when a movie is started the Netflix app opens and plays the video. Is that possible with Netflix?

I‘m asking, because there is an option like that for the Amazon Prime VOD addon (Sandman).
But Amazon may work differently altogether.

either way, thank you for your great work!!

**Comment by CastagnaIT** [comment]
@giddybitty Because of the many problems that Netflix is causing, and since after months no one has been able to help me in any way to improve the situation, i decided to keep the addon working as much as possible, from my part i will not invest time to introduce new features

for those wondering what is going on and what the problem is, I again point out that the Issue #1627 has been open for some time now that explains things in detail, so updates, if any, will be posted on  Issue #1627

**Comment by andrasj** [comment]
Hello, probably you guys already went down this road, but do we have the answer to: "why is it failing on specific OS's only?"
Why does it work on linux? Isn't there a renew request for the license? afaik NF-servers shouldn't see the difference between linux/windows/... clients, if the requests are generated by the same code. (or are these made from widevine?)
Did anyone already dump (decode/compare/...?) the traffic of different working/non-working OS-versions?

I didn't know windows was broken for this long... I've got an old box with openELEC which is playing NF fine... I encountered this while supporting a friend who was trying NF on windows...
(FYI: I'm a developer, when I encounter these kind of issues, I prefer to look/understand what happens on the wire. Unfortunately I'm not familliar with the various DRM-flavours).

**Comment by Cuplight** [comment]
A few days ago this addon worked correctly for me again on Android with Kodi 21.1. For months I had the problem that after 2 minutes the playback stopped.

**Comment by tyshiro** [comment]
Is it still working now?Sent from my iPhoneOn Aug 30, 2024, at 05:52, Cuplight ***@***.***> wrote:﻿
A few days ago this addon worked correctly for me again on Android with Kodi 21.1. For months I had the problem that after 2 minutes the playback stopped.

**Comment by Cuplight** [comment]
> Is it still working now?Sent from my iPhoneOn Aug 30, 2024, at 05:52, Cuplight ***@***.***> wrote:﻿ A few days ago this addon worked correctly for me again on Android with Kodi 21.1. For months I had the problem that after 2 minutes the playback stopped. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

Currently for me yes.

**Comment by Cuplight** [comment]
After 2 days working correctly with 1080p resolution I can now only play more than 2 minutes with SD 576p.

**Comment by tyshiro** [comment]
At least you still get that I still can’t play anything.Sent from my iPhoneOn Sep 1, 2024, at 13:39, Cuplight ***@***.***> wrote:﻿
After 2 days working correctly with 1080p resolution I can now only play more than 2 minutes with SD 576p.

**Comment by grapenose** [comment]
I'm now getting a message that Due to a MSL error It is no longer possible to use the Netflix Add on. However I was able to use Netflix on my Sony TV up until a few days ago when this message started appearing. 
Has something changed or is this message inserted for  anyone trying to use the add on.

**Comment by adamhotep** [comment]
This started affecting my Linux install at some point in the last 24h (I watched something successfully last night, could not tonight). Everything works for the first 2:01.5, then audio is fine but there is no video due to the encryption scheme noted above.

**Sad workaround:** Go to Netflix → Settings → Expert → Playback and set `Limit video stream resolution to` = `SD 576p`

Netflix add-on 1.23.3 (also true for 1.23.2 and 1.23.1), with and without VP9 enabled, Kodi 20.2.0 (stuck on Ubuntu [PPA](https://kodi.tv/article/ubuntu-team-kodi-ppa-officially-retired/), refusing to use flatpak, will eventually reinstall with Debian to use the deb-multimedia package).

**Comment by adamhotep** [comment]
Update: as of today, I had to lower it to 480p for the workaround.

**Comment by nonobio** [comment]
https://github.com/CastagnaIT/plugin.video.netflix/issues/1627#issuecomment-2824297160

Hi, are there alternative solutions outside Kodi?

**Comment by adamhotep** [comment]
This issue came back for me today. Even downgrading to 480p on both Netflix and InputStream Adaptive fails to get past the first 2min, with "This title is not available to watch instantly. Please try another title."

This is on amd64.

**Comment by jankofron** [comment]
Just to update with my experience. My suspicion is that Netflix is doing something on their side, as one day, I can play everything in HD, another day just 576p (otherwise got this "MSL: The title is not available to play instantly..." message and no video after those 2 minutes). My boxes are Linux/amd64 (Ubuntu 24.04 and Arch, both Kodi 21.2 + NF addon 1.23.4+matrix.1). Today (May 16, 2025), it works for me HD. I have set fixed video resolution (and set it to 1080HD) in the NF addon Expert settings and have not touched InputStreamAdaptive or other settings. I can share further details, if anybody is interested.

**Comment by lameventanas** [comment]
I'm experiencing the opposite problem: the first 2 minutes can't be played, but if I seek forward beyond 2 minutes, everything works.
Tried with different resolutions and even reset to the default addon settings, but nothing.

**Comment by hunaqua** [comment]
> Én pont az ellenkező problémával küzdök: az első 2 percet nem lehet lejátszani, de ha 2 percen túl előretekerek, minden működik. Próbáltam különböző felbontásokkal, sőt, még az alapértelmezett kiegészítő beállításokra is visszaállítottam, de semmi.

Unfortunately, the same error. I can't find a solution.

**Comment by dkambur** [comment]
We had the same problem. None of the suggested workarounds seemed to work. Downgrading Netfix to most basic plan seems to have provided a workaround.

**Comment by Marietto2008** [comment]
I've installed kodi + the netflix addon on ubuntu 24.04. I can't play any video due to the error : playback failed. One or more items failed to play. Check the log for more information about this message. Where is the log saved ?

**Comment by jankofron** [comment]
> I've installed kodi + the netflix addon on ubuntu 24.04. I can't play any video due to the error : playback failed. One or more items failed to play. Check the log for more information about this message. Where is the log saved ?

@Marietto2008 Currently, it seems that unless you have saved a working login (Widevine ESN ID), you cannot create a new working Netflix session from the plugin. There are people (at least me :-) ) trying to find out what has changed at the Netflix side, but no solution seems to be out there, so far.

The log is available under ~/.kodi/temp/kodi.log under linux.

**Comment by Marietto2008** [comment]
I will fix this bug,but only for kodi installed on FreeBSD.
